### PR TITLE
fix(integrity): expect pkg mgmt files

### PR DESCRIPTION
**Pull request by blizzz**
backport of #488

### CODE REVIEW

**Comment by blizzz** [comment]
Tested on an instance that was having the issue, and it fixes it.

**Review by AndyScherzinger — APPROVED** [review]

**Review by juliushaertl — APPROVED** [review]

**Comment by come-nc** [comment]
CI is not happy.

When doing "composer i" I end up with files removed from git, is that expected?

**Comment by blizzz** [comment]
> CI is not happy.
> 
> When doing "composer i" I end up with files removed from git, is that expected?

Sounds strange, what would it want to remove? 

The makefile (and also tests) differ from master, it surely is worth to streamline them as well.

**Comment by blizzz** [comment]
Pushed minimal adjustments to the tests.

**Comment by blizzz** [comment]
"minimal"… tbc

**Comment by blizzz** [comment]
totally nuts. Why don't I copy over test definition from master? It succeeds even on 8.2 for unknown reasons, but it does not test against 27. Still might give it a try tomorrow, this is frustration. In this repo we almost never have PRs against stable branches…

File: tests/features/master.feature
Comment (come-nc) [inline tests/features/master.feature:1]: Feels weird to stop testing that, support for daily channel is a feature of the updater.

File: tests/features/stable20.feature [hunk 1]
```diff
@@ -1,54 +0,0 @@
...
-    Given the current installed version is 20.0.0beta1
-    And there is an update to prerelease version "20.0.0RC1" available
-    And the version number is decreased in the config.php to enforce upgrade
-    When the CLI updater is run successfully
-    And the output should contain "Update successful"
-    Then the installed version should be 20.0
-    And maintenance mode should be off
-    And upgrade is not required
-
-  Scenario: Update is available but unexpected folder found - 20.0.7 to 20.0.8
```
Comment (come-nc) [inline tests/features/stable20.feature:13]: unexpected folder here.

All these tests should be kept IMO since they test features of the updater.

**Review by come-nc — CHANGES_REQUESTED** [review]

**Comment by come-nc** [comment]
Also why the symfony/console bump in stable27 that was not done in master first?

**Comment by come-nc** [comment]
> > CI is not happy.
> > When doing "composer i" I end up with files removed from git, is that expected?
> 
> Sounds strange, what would it want to remove?
> 
> The makefile (and also tests) differ from master, it surely is worth to streamline them as well.

My composer was outdated, this does not happen anymore.

**Comment by blizzz** [comment]
> Also why the symfony/console bump in stable27 that was not done in master first?

Tests result complained about PHP 8.2 incompatibilities, so that was a try. That's also why I wonder why they work in master.

**Comment by blizzz** [comment]
As discussused wiht @come-nc  we'll use  https://github.com/nextcloud/updater/pull/492 for an immediate fix with green CI and he'll do an  follow up with freshing up tests. The fundamental techniques of the updater are still tested, there is only very little interaction with business logic, and master still succeeds.

**Pull request closed without merging**